### ISSUE TITLE
Sonos Documentation

### ISSUE THREAD

**Issue by Vitchling** [openhab/openhab#3847]
Hi

I've successfully implemented a play/pause control for Sonos on my 1.7.0 OH instance but I have no idea how to play a favorite or playlist and the documentation on this is weak. Can anyone please provide an example sitemap, items and rule for how I can play a favorite or playlist under program control?

Cheers
Andrew

**Comment by luotaus** [comment]
Hi Andrew

not quite sure what you mean with "weak documentation", the [wiki] (https://github.com/openhab/openhab/wiki/Sonos-Binding) has an example item/sitemap for favorites. 

Please also have a look at the forum https://community.openhab.org/ which has some threads around SONOS. 

Suggest you close this issue  and use a discussion in the forum (where you most likely get more help than here).

**Comment by teichsta** [comment]
> Please also have a look at the forum https://community.openhab.org/ which has some threads around SONOS.

+1

**Comment by Vitchling** [comment]
Thank you
 
As a novice and not a programmer, I find that an Item definition, sitemap
entry and a rule to make it work is very useful. I'm smart enough to adapt
and re-use code that works but I'm not smart enough to realise that a
command sent to a String Item needs commas around it.
 
e.g.    sendCommand("Sonosfavorite" , "BurglarAlarm")
 
rather than   sendCommand(Sonosfavorite , "BurglarAlarm")
 
or   sendCommand(Sonosfavorite , BurglarAlarm)
 
This is why I feel that the documentation is weak. It presumes that the
people reading it are programmers and that the only requirement of the
documentation is to explain the available functions of the device.
 
I'm sure that I am not the only person that uses openHAB that is not an
experienced programmer. There are plenty of posts from people that are
trying to learn but struggle with what to you is simple syntax.
 
I'm not a programmer but I am smart enough to have created a ZWave heating
system that controls my boiler and each individual room based on a Gcal
schedule. It now speaks to me when I set or unset my Alarm and if the alarm
is triggered it now plays quite a scary message to any burglar.
 
When I was a young engineer, one of my jobs was to write documentation for
those that would come after me. I never assumed that they knew anything but
I did assume that they could learn by example and they did. As I do now,
with openHAB.
 
A device like Sonos is complex and has many functions. People will want to
use it in different ways. I would have benefited from a simple rule example
such as this:
 
sendCommand("Sonosfavorite" , "BurglarAlarm")
 
It would have taught me that a string item needs quote marks.
 
Obvious to you, impossible to me. 
 
I have working examples of Sonos Items and rules that would maybe help
others that want to use Favourites. Maybe you could find a way to update the
Sonos wiki to include them.
 
Items
String Sonosfavorite "Favorite [%s]" (Sonos) {sonos="[DiningRoom:favorite]",
autoupdate="false"}
 
Rules
sendCommand("Sonosfavorite" , "BurglarAlarm")
 
Cheers
Vitch
 
 
 
 

  _____

**Comment by Vitchling** [comment]
Thank you.
 
I resolved my problem by reading around string items. It was a simple syntax
problem in the end. I do think the Sonos wiki would benefit from an item and
rule example that shows how to play a Favourite though.
 
e.g. 
 
String Sonosfavorite "Favorite [%s]" (Sonos) {sonos="[DiningRoom:favorite]",
autoupdate="false"}
 
sendCommand("Sonosfavorite" , "BurglarAlarm")
 
I was buggered by not knowing that I had to use quotation marks. It took me
a few days to work that simple thing out because there was nothing in the
Sonos wiki to indicate that. Hey ho, I'm not a programmer so these are
inevitable setbacks.
 
Cheers
Andrew

  _____
